api/libproject/devstart.rb

```diff
@@ -370,10 +370,26 @@ def run_public_api_tests(cmd_name, args)
   :fn => ->(*args) { run_public_api_tests("test-public-api", args) }
 })
 
+def test_api_changes(branch_name)
+  common = Common.new
+  common_commit = common.capture_stdout(%W{git merge-base master blrubenstein/test-circle}).split("\n")[0]
+  api_lines = common.capture_stdout(%W{git diff --name-only #{common_commit}}).split("\n")
+  return api_lines.any? { |s| s.include?('api') }
```

Comment:
This substring check feels like it should be tightened up to match a directory specifically, e.g. via startswith or including "/"s. Note that specifically in this case we care about "api" and "common-api".